v3_usm.go

```diff
@@ -622,28 +622,89 @@ func (sp *UsmSecurityParameters) discoveryRequired() *SnmpPacket {
 }
 
 func (sp *UsmSecurityParameters) calcPacketDigest(packet []byte) []byte {
-	var mac hash.Hash
+	return calcPacketDigest(packet, sp)
+}
 
-	switch sp.AuthenticationProtocol {
-	default:
-		mac = hmac.New(crypto.MD5.New, sp.SecretKey)
-	case SHA:
-		mac = hmac.New(crypto.SHA1.New, sp.SecretKey)
-	case SHA224:
-		mac = hmac.New(crypto.SHA224.New, sp.SecretKey)
-	case SHA256:
-		mac = hmac.New(crypto.SHA256.New, sp.SecretKey)
-	case SHA384:
-		mac = hmac.New(crypto.SHA384.New, sp.SecretKey)
-	case SHA512:
-		mac = hmac.New(crypto.SHA512.New, sp.SecretKey)
-	}
+// calcPacketDigest calculate authenticate digest for incoming messages (TRAP or
+// INFORM).
+// Support MD5, SHA1, SHA224, SHA256, SHA384, SHA512 protocols
+func calcPacketDigest(packetBytes []byte, secParams *UsmSecurityParameters) []byte {
```

Comment:
The ``isAuthentic()`` function where ``calcPacketDigest`` is called can handle errors. 
Could you update the function to look as follow catch and return errors happening in the ``digest*()`` functions:

```go
func calcPacketDigest(packetBytes []byte, secParams *UsmSecurityParameters) ([]byte, error) {
	var digest []byte
	var err error

	switch secParams.AuthenticationProtocol {
	case MD5, SHA:
		digest, err = digestRFC3414(
			secParams.AuthenticationProtocol,
			packetBytes,
			secParams.SecretKey)
	case SHA224, SHA256, SHA384, SHA512:
		digest, err = digestRFC7860(
			secParams.AuthenticationProtocol,
			packetBytes,
			secParams.SecretKey)
	}

	return digest, err
}
```